scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowDoFns.scala

```diff
@@ -18,31 +18,29 @@
 package com.spotify.scio.tensorflow
 
 import java.time.Duration
-import java.util.concurrent.{ConcurrentHashMap, ConcurrentMap}
-import java.util.function.Function
 
-import com.spotify.scio.transforms.DoFnWithResource
-import com.spotify.scio.transforms.DoFnWithResource.ResourceType
-import com.spotify.zoltar.tf.TensorFlowModel
+import com.spotify.zoltar.tf.{TensorFlowLoader, TensorFlowModel}
 import com.spotify.zoltar.{Model, Models}
 import org.apache.beam.sdk.transforms.DoFn
-import org.apache.beam.sdk.transforms.DoFn.{ProcessElement, Teardown}
-import org.tensorflow._
+import org.apache.beam.sdk.transforms.DoFn.{ProcessElement, Setup, Teardown}
 import org.slf4j.LoggerFactory
-
+import org.tensorflow._
+import org.tensorflow.example.Example
 import scala.collection.JavaConverters._
 
-private[this] abstract class PredictDoFn[T, V, M <: Model[_]](
-  fetchOp: Seq[String],
-  inFn: T => Map[String, Tensor[_]],
-  outFn: (T, Map[String, Tensor[_]]) => V
-) extends DoFnWithResource[T, V, ConcurrentMap[String, M]] {
+import com.spotify.zoltar.Model.Id
+
+private[this] abstract class PredictDoFn[T, V, M <: Model[_]] extends DoFn[T, V] {
```

Comment:
I like the idea of refactoring this by moving the arguments to methods;

we can make this a `sealed trait`